frontend/src/components/Home/ItemCard.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import Card from '@mui/material/Card';
+import CardActions from '@mui/material/CardActions';
+import CardContent from '@mui/material/CardContent';
+import { Typography, Grid, Button, CardHeader, CardMedia } from '@mui/material';
+
+class ItemCard extends Component {
+    constructor(props) {
+        super(props);
+    }
+
+    // is it a joke? an image? a taunt? return the appropriate card
+    _resolveItemType = (type, payload) => {
```

Comment:
Is there any actual value in having this generic ItemCard that resolves to more specific types?
Otherwise wouldn't it make more sense to just have MainContent or wherever we have that shows data just figure out what card types they should use directly.

I guess unless ItemCard is going to have a decent amount of generic capability it just seems redundant. But unclear if that's the "normal" way to do things in React.

Reply:
I believe I was expecting to have an endpoint that returned most recent items (jokes, images, taunts) hence trying to create a component that can accommodate for all types. Since we don't have that it will make more sense to just create specific card types. I will change this accordingly.